src/compiler/check/writer.ts

```diff
@@ -0,0 +1,761 @@
+/// <reference path="../binder.ts" />
+
+/* @internal */
+namespace ts {
+    export function createWriter(
```

Comment:
"Writer" is a pretty generic name, but I guess it doesn't write any one thing. `SemanticWriter`?